### PR TITLE
Prefix identifiers in stubs with function name

### PR DESCRIPTION
**Pull request by peterschrammel**
The name of an auxiliary variable must not depend on other functions in the symbol table.
The function_id field in the object_factory_parameters wasn't always correctly set.
This breaks goto-diff.

### CODE REVIEW

**Comment by smowton** [comment]
Note test-gen will need to apply the same change to its more complex java-opaque-method-stubs code.

**Review by thk123 — COMMENTED** [review]
A unit test (or a regression test using `--show-goto-functions`) would be a good idea here. 

Needs a TG pointer bump as @smowton pointed out.

**Comment by peterschrammel** [comment]
> A unit test (or a regression test using --show-goto-functions) would be a good idea here.

Would require using --show-symbol-table because --show-goto-functions only shows the prettified name.

**Comment by peterschrammel** [comment]
@thk123, @smowton, I tried to deal with both of your comments. Any further suggestions welcome.

File: jbmc/regression/jbmc/stub_identifiers/test.desc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+^Symbol\.*: to_return
+^Symbol\.*: this
+^Symbol\.*: caught_exception
+^Symbol\.*: anonlocal
+^Symbol\.*: stub_ignored_arg
+^Symbol\.*: #return_value
+^Symbol\.*: return_tmp
+^Symbol\.*: new_tmp
+^Symbol\.*: tmp_new_data_array
+^Symbol\.*: newarray_tmp
```
Comment (smowton) [inline jbmc/regression/jbmc/stub_identifiers/test.desc:20]: Add positive conditions too? At the moment `main() { return 0; }` would pass this test-case ;)

File: jbmc/regression/jbmc/stub_identifiers/test.desc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+CORE symex-driven-lazy-loading-expected-failure
```
Comment (smowton) [inline jbmc/regression/jbmc/stub_identifiers/test.desc:1]: Add a comment regarding why this doesn't work with symex-driven loading

**Review by thk123 — COMMENTED** [review]
Changes look good (though I agree with Chris's comments) - but the TG pointer bump is currently failign

File: jbmc/regression/jbmc/stub_identifiers/test.desc (hunk 1 above)
Comment (peterschrammel) [reply jbmc/regression/jbmc/stub_identifiers/test.desc:20]: That would be perfectly fine. I only want to check that no non-prefixed variables are generated - these break goto-diff. I added two checks if this is preferred.

**Comment by peterschrammel** [comment]
@thk123, @smowton, I have taken into account your comments. TG bump is passing.

**Review by thk123 — APPROVED** [review]
TG bump passing

**Review by smowton — APPROVED** [review]

**Pull request merged**